### PR TITLE
 Cramér GAN

### PR DESCRIPTION
**Pull request by martyn**
New loss function.
https://arxiv.org/pdf/1705.10743.pdf

Remaining:
* [x] Samples
* [ ] Investigate f(x) or hyperparams
* [ ] Remove hacks

### CODE REVIEW

**Comment by martyn** [comment]
image

Different samples.  Not quite working yet.
